### DISCUSSION TITLE
Is there a way to manipulate the HTTP headers before uploading a file?

### DISCUSSION THREAD

**Discussion by plamber** [rpldy/react-uploady#141] [Q&A]
**Is your feature request related to a problem? Please describe.**
I was looking for a way to manipulate the header information before uploading a file. We have an API that requires a Baerer token for authentication. Before triggering the upload process I would need to retrieve the baerer token and update the header information before posting the file.

Unfortunately, I haven't found an example how you can do this.

Thank you very much for your feedback

**Comment by yoavniran** [comment]
moving to [discussions](https://github.com/rpldy/react-uploady/discussions). will answer there

**Comment by yoavniran** [comment]
Yes, this is definitely possible.
The [Guides](https://github.com/rpldy/react-uploady/blob/master/guides) section has many useful tutorials on how to achieve more advanced flows with Uploady.
This in particular is what you'll need: https://github.com/rpldy/react-uploady/blob/master/guides/DynamicParameters.md#using-hook

In your case, you can retrieve the info you need from the `useRequestPreSend` hook callback (it can be async) and then return an object similar to this:

```javascript
  return {
    destination: {
         headers: {
              "my-header": "header-value"
          }
    } 
  };
```

The content of the object you return from the hook will be merged with the other upload options.

**Reply by plamber** [reply]
Thank you for the feedback. Will give it a try
